BedrockServer.cpp

```diff
@@ -863,13 +863,16 @@ void BedrockServer::runCommand(unique_ptr<BedrockCommand>&& _command, bool isBlo
                 }
             }
 
-            uint64_t conflictLockStartTime = 0;
-            if (lastConflictPage) {
-                conflictLockStartTime = STimeNow();
-            }
-            PageLockGuard pageLock(lastConflictPage);
-            if (lastConflictPage) {
-                SINFO("Waited " << (STimeNow() - conflictLockStartTime) << "us for lock on db page " << lastConflictPage << ".");
+            {
+                BedrockCore::AutoTimer timer(command, BedrockCommand::QUEUE_PAGE_LOCK);
+                uint64_t conflictLockStartTime = 0;
+                if (lastConflictPage) {
+                    conflictLockStartTime = STimeNow();
+                }
+                PageLockGuard pageLock(lastConflictPage);
+                if (lastConflictPage) {
+                    SINFO("Waited " << (STimeNow() - conflictLockStartTime) << "us for lock on db page " << lastConflictPage << ".");
+                }
```

Comment:
Is this on purpose nested in `{ ... }`?

Reply:
Is it so the AutoTimer is destroyed when we leave this block?

Reply:
Yes, which brings up a great point - I think this will also cause `pageLock` to be destroyed and unlocked after this block exits, which we do not want...

Reply:
@aldo-expensify yes

Reply:
@rafecolton I think you're right... what should we do instead? Put this timer inside the `PageLockGuard` constructor?

Reply:
Or declare it outside the block

Reply:
That's not going to work, because basically I want to time how long we are spending in the constructor 😄 An alternative would be to move the locking code outside of the constructor I suppose

Reply:
```suggestion
                BedrockCore::AutoTimer *timer = new BedrockCore::AutoTimer(command, BedrockCommand::QUEUE_PAGE_LOCK);
                uint64_t conflictLockStartTime = 0;
                if (lastConflictPage) {
                    conflictLockStartTime = STimeNow();
                }
                PageLockGuard pageLock(lastConflictPage);
                if (lastConflictPage) {
                    SINFO("Waited " << (STimeNow() - conflictLockStartTime) << "us for lock on db page " << lastConflictPage << ".");
                }
                delete timer;
```

Reply:
Yeah that would work too 👍

Reply:
updated